frontend/packages/ceph-storage-plugin/integration-tests/utils/helpers.ts

```diff
@@ -114,3 +114,67 @@ export const refreshIfNotVisible = async (element, maxTimes = 1) => {
     count += 1;
   }
 };
+
+export const getIds = (nodes: nodeType[], type: string): number[] =>
+  nodes.filter((node) => node.type === type).map((node) => node.id);
+
+export const getNewOSDIds = (nodes: nodeType[], osds: number[]): number[] =>
+  nodes.filter((node) => node.type === OSD && osds.indexOf(node.id) === -1).map((node) => node.id);
+
+// created dictionary for faster acess O(1)
+export const createOSDTreeMap = (nodes: nodeType[]): osdTreeType => {
+  const tree = {};
+  nodes.forEach((node) => {
+    tree[node.id] = node;
+  });
+  return tree;
+};
+
+export const verifyZoneOSDMapping = (
+  zones: number[],
+  osds: number[],
+  osdtree: osdTreeType,
+): boolean => {
+  let filteredOsds = [];
+  zones.forEach((zone) => {
+    const hostId = osdtree[zone].children[0];
+    const len = osdtree[hostId].children.length;
+    filteredOsds = osds.filter((osd) => osd !== osdtree[hostId].children[len - 1]);
+  });
+
+  return filteredOsds.length === 0;
+};
+
+export const verifyNodeOSDMapping = (
+  nodes: number[],
+  osds: number[],
+  osdtree: osdTreeType,
+): boolean => {
+  let filteredOsds = [];
+  nodes.forEach((node) => {
+    const len = osdtree[node].children.length;
+    filteredOsds = osds.filter((osd) => osd !== osdtree[node].children[len - 1]);
+  });
+
+  return filteredOsds.length === 0;
+};
+
+export type nodeType = {
```

Comment:
just for differentiating between type names and var/key names/types
```suggestion
export type Node = {
```

Reply:
Ack for the Pascal case. Whats the reason for omitting `type`. Its being added to differentiate its type name from the variable name

Reply:
You can add if you want, defining in Pascal works  thats why I dont encourage.